umm-spec-lib/src/cmr/umm_spec/test/echo10_expected_conversion.clj

```diff
@@ -259,6 +259,13 @@
       [(cmn/map->ResourceCitationType
         {:OtherCitationDetails other-citation-details})])))
 
+(defn- expected-doi
+  "Updates the expected DOI for ECHO10."
+  [doi]
+  (let [updated-doi (util/remove-nil-keys (dissoc doi :MissingReason :Explanation))]
```

Comment:
Nitpick: Would a `when-let` work here? I see this `let -> when -> seq` in a few places

Reply:
`updated-doi` is a map, not a collection. I was checking if it was empty (had no keys) with the `seq`. I don't actually want to return the seq of a map, so I couldn't use a `when-let`.